src/key/helper.js

```diff
@@ -269,10 +266,11 @@ export async function isDataRevoked(primaryKey, signatureType, dataToVerify, rev
         // third-party revocation signatures here. (It could also be revoking a
         // third-party key certification, which should only affect
         // `verifyAllCertifications`.)
-        (!signature || revocationSignature.issuerKeyID.equals(signature.issuerKeyID)) &&
-        !(config.revocationsExpire && revocationSignature.isExpired(normDate))
+        !signature || revocationSignature.issuerKeyID.equals(signature.issuerKeyID)
       ) {
-        revocationSignature.verified || await revocationSignature.verify(key, signatureType, dataToVerify, undefined, config);
+        revocationSignature.verified || await revocationSignature.verify(
```

Comment:
Just noticed that if the revocation signature expires soon after we have verified it once, the signature is "always" considered verified regardless, due to the caching. The same issue is present wherever `.verified` is checked in the rest of the code.
I still think it's a good idea to check `isExpired` as part of `verify`, so to solve the problem, I'd rather check `.verified` inside `verify` as well, and then add a new optional parameter `reverify=true` to determine whether to discard the cached value (something similar was already proposed in #880).

Reply:
I just added a test to detect this problem -- it's failing for now, as expected.

Reply:
I agree moving the `.verified` check into `verify` makes sense, but I'd prefer not to have a `reverify` parameter, rather I would just consider `.verified` to only mean that the signature was cryptographically valid, but always re-check expiry (it should be very cheap to do so anyway).